File: backend/schemas.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel, Field

class CreateAuditRequest(BaseModel):
    name: str = Field(min_length=1)
    platform: Literal["mobile-web", "desktop-web", "app"]

class ScreenDto(BaseModel):
    id: str
    order: int
    flowStep: str
    imageUrl: str
    findingCount: int = 0

class FindingDto(BaseModel):
    id: str
    ruleId: Literal["DA-03", "DA-04", "DA-12", "DA-15"]
    riskType: Literal["PRESELECTED_OPTION", "VISUAL_HIERARCHY_DISTORTION", "EMOTIONAL_LANGUAGE", "SEQUENTIAL_PRICE_DISCLOSURE"]
    title: str
    description: str
    screenIds: list[str]
    element: str
    defaultState: str | None = None
    costImpact: str | None = None
    severity: Literal["HIGH", "REVIEW"]
    status: Literal["open", "reviewing", "resolved"] = "open"
    confidence: float = Field(ge=0, le=1)
    recommendation: str
    guideline: str

class AuditDto(BaseModel):
    id: str
    name: str
    platform: Literal["mobile-web", "desktop-web", "app"]
    status: Literal["draft", "queued", "analyzing", "completed", "failed"]
    updatedAt: datetime
    screens: list[ScreenDto]
    findings: list[FindingDto]

class JobDto(BaseModel):
    jobId: str
    auditId: str
    status: Literal["queued", "analyzing", "completed", "failed"]
    progress: float = Field(ge=0, le=100)

class FindingStatusRequest(BaseModel):
    status: Literal["open", "reviewing", "resolved"]

File: frontend/src/api/schemas.ts
import { z } from "zod";

export const findingSchema = z.object({
  id: z.string().min(1),
  ruleId: z.enum(["DA-03", "DA-04", "DA-12", "DA-15"]),
  riskType: z.enum([
    "PRESELECTED_OPTION",
    "VISUAL_HIERARCHY_DISTORTION",
    "EMOTIONAL_LANGUAGE",
    "SEQUENTIAL_PRICE_DISCLOSURE",
  ]),
  title: z.string(),
  description: z.string(),
  screenIds: z.array(z.string()),
  element: z.string(),
  defaultState: z.string().optional(),
  costImpact: z.string().optional(),
  severity: z.enum(["HIGH", "REVIEW"]),
  status: z.enum(["open", "reviewing", "resolved"]),
  confidence: z.number().min(0).max(1),
  recommendation: z.string(),
  guideline: z.string(),
});

export const auditSchema = z.object({
  id: z.string().min(1),
  name: z.string().min(1),
  platform: z.enum(["mobile-web", "desktop-web", "app"]),
  status: z.enum(["draft", "queued", "analyzing", "completed", "failed"]),
  updatedAt: z.string().datetime({ offset: true }),
  screens: z.array(
    z.object({
      id: z.string().min(1),
      order: z.number().int().positive(),
      flowStep: z.string().min(1),
      imageUrl: z.string(),
      findingCount: z.number().int().nonnegative(),
    }),
  ),
  findings: z.array(findingSchema),
});

export const dashboardSummarySchema = z.object({
  activeAuditId: z.string().nullable(),
  audits: z.array(auditSchema),
});

export const analysisJobSchema = z.object({
  jobId: z.string().min(1),
  auditId: z.string().min(1),
  status: z.enum(["queued", "analyzing", "completed", "failed"]),
  progress: z.number().min(0).max(100),
});
